### PR TITLE
Added support for rails5 and rspec 3.5

### PR DESCRIPTION
**Pull request by sivagollapalli**
#97

### CODE REVIEW

File: rspec-sidekiq.gemspec [hunk 1]
```diff
@@ -11,13 +11,13 @@ Gem::Specification.new do |s|
...
 
-  s.add_dependency 'rspec-core', '~> 3.0', '>= 3.0.0'
+  s.add_dependency 'rspec-core', '~> 3.5', '>= 3.5.0'
   s.add_dependency 'sidekiq', '>= 2.4.0'
 
   s.add_development_dependency 'coveralls', '~> 0.8', '>= 0.8.0'
   s.add_development_dependency 'fuubar', '~> 2.0', '>= 2.0.0'
-  s.add_development_dependency 'activejob', '~> 4.2', '>= 4.0.0'
-  s.add_development_dependency 'actionmailer', '~> 4.2', '>= 4.0.0'
+  s.add_development_dependency 'activejob', '~> 5.0', '>= 5.0.0'
```
Comment (philostler) [inline rspec-sidekiq.gemspec]: Does Sidekiq still support 4.x?

**Comment by philostler** [comment]
Could you please rebase this?

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ rvm:
   - 2.0
   - 2.1
   - 2.2
+  - 2.3.0
```
Comment (mathieujobin) [inline .travis.yml]: 2.3.1 instead?

File: .travis.yml [hunk 3]
```diff
@@ -11,3 +12,6 @@ before_install:
 matrix:
   allow_failures:
     - rvm: rbx-2
+    - rvm: jruby
+    - rvm: 2.0
+    - rvm: 2.1
```
Comment (mathieujobin) [inline .travis.yml]: Why should these 3 suddenly fail?

File: rspec-sidekiq.gemspec [hunk 4]
```diff
@@ -11,13 +11,13 @@ Gem::Specification.new do |s|
   s.description = 'Simple testing of Sidekiq jobs via a collection of matchers and helpers'
   s.license     = 'MIT'
 
-  s.add_dependency 'rspec-core', '~> 3.0', '>= 3.0.0'
+  s.add_dependency 'rspec-core', '~> 3.5', '>= 3.5.0'
```
Comment (mathieujobin) [inline rspec-sidekiq.gemspec]: This line did not need to change. 3.5 was not blocked. And now 3.5 is required?

File: .travis.yml (hunk 3 above)
Comment (sivagollapalli) [inline .travis.yml]: Since `rails 5` only runs `ruby 2.2` and above. So I added here.

File: rspec-sidekiq.gemspec (hunk 1 above)
Comment (sivagollapalli) [inline rspec-sidekiq.gemspec]: Looks like I didn't get you. Could you elaborate?

**Comment by sivagollapalli** [comment]
@philostler Could you provide your comments?

**Comment by philostler** [comment]
@sivagollapalli Sure, hard for me as I'm quite distant from Ruby/Rails these days.
Could you help out with understanding the change.
* Rails 5 appears to support Ruby 2.2.2+ so why is the branch code at 2.3.0?
* How does this add RSpec 3.5 support?
* If I wanted to run Ruby 2.3.0+ with Rails 4, wouldn't this change break?

**Comment by packrat386** [comment]
Instead of pinning rails to a different version based on which ruby you're running, you should have different gemfiles for running tests against different versions. See [here](https://github.com/mgomes/api_auth/blob/master/.travis.yml) for an example.

**Comment by philostler** [comment]
^^What he says ;) Of course! I've spend too long in JavaScript land to remember that pattern

**Comment by sivagollapalli** [comment]
@philostler Instead of checking `ActionMailer::VERSION::STRING >= "5.0.0"` I have checked ruby version. So it breaks if a user runs Ruby 2.3.0+ with Rails 4. I will modify that as following

```
 if ActionMailer::VERSION::STRING >= "5.0.0" || ActiveJob::VERSION::STRING >= "5.0.0"
    s.add_dependency 'rspec-core', '~> 3.5', '>= 3.5.0'

    s.add_development_dependency 'activejob', '~> 5.0', '>= 5.0.0'
    s.add_development_dependency 'actionmailer', '~> 5.0', '>= 5.0.0'
    s.add_development_dependency 'rspec', '~> 3.5'
  else
    s.add_dependency 'rspec-core', '~> 3.0', '>= 3.0.0'

    s.add_development_dependency 'activejob', '~> 4.2', '>= 4.0.0'
    s.add_development_dependency 'actionmailer', '~> 4.2', '>= 4.0.0'
    s.add_development_dependency 'rspec', '~> 3.0'
  end
```
But the problem with above code is I couldn't get `ActionMailer::VERSION::STRING` value. Could you help me?

**Comment by packrat386** [comment]
@sivagollapalli you're trying to see which dependency to install by checking which dependency is installed, which isn't going to work. Your development dependency should be as follows.
```ruby
s.add_development_dependency 'activejob', '>= 4.0.0'
s.add_development_dependency 'actionmailer', '>= 4.0.0'
s.add_development_dependency 'rspec', '~> 3.0'
```
and then if you want to test against against different versions of activejob etc., you can use multiple gemfiles as in [this example](https://github.com/mgomes/api_auth/blob/master/.travis.yml)

The runtime dependency on rspec should be as permissive as possible so:
```ruby
s.add_dependency 'rspec-core', '~> 3.0'
```
again, if you want to test against rspec 3.5 in particular, you can use a separate gemfile to do it.

**Pull request closed without merging**